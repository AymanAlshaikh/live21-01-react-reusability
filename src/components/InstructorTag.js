import { GoToGithub, Wrapper, Emoji, Name } from "./styles";

const InstructorTag = ({ emoji, name, github }) => {
  const goToGitHubPage = () => window.open(`https://github.com/${github}`);

  return (
    <Wrapper onClick={goToGitHubPage}>
      <Emoji>{emoji}</Emoji>
      <Name>{name}</Name>
      <GoToGithub>Go to GitHub</GoToGithub>
    </Wrapper>
  );
};

export default InstructorTag;
