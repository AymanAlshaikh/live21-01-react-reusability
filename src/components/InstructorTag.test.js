import { fireEvent, render } from "@testing-library/react";
import InstructorTag from "./InstructorTag";
import "jest-styled-components";

describe("InstructorTag component", () => {
  // jest-styled-components doesn't allow render to work in descrive closures
  // thus, I have created a separate new render of the component every time

  window.open = jest.fn();

  test("renders correctly", () => {
    const { container } = render(
      <InstructorTag emoji="ðŸŒ²" name="EglÄ—" github="egloo8" />
    );

    expect(container).toMatchSnapshot();
  });

  test("opens github account on click", () => {
    const { getByText } = render(
      <InstructorTag emoji="ðŸŒ²" name="EglÄ—" github="egloo8" />
    );

    const goToGitGubPage = getByText("Go to GitHub");
    fireEvent.click(goToGitGubPage);
    expect(window.open).toHaveBeenCalledWith("https://github.com/egloo8");
  });
});
