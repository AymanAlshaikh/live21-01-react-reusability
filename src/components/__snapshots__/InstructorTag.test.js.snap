// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstructorTag component renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 500px;
  height: 80px;
  border-radius: 300px;
  padding: 0px 25px;
  margin-bottom: 25px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: rgba(0,0,0,0.4);
}

.c0:hover {
  background-color: rgba(0,0,0,0.8);
}

.c1 {
  margin-right: 20px;
  font-size: 1.6em;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-size: 1.6em;
  font-weight: 500;
  color: rgba(255,255,255,0.9);
}

.c3 {
  font-size: 0.8em;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: rgba(255,255,255,0.3);
}

<div>
  <div
    class="c0"
  >
    <span
      class="c1"
    >
      ðŸŒ²
    </span>
    <span
      class="c2"
    >
      EglÄ—
    </span>
    <span
      class="c3"
    >
      Go to GitHub
    </span>
  </div>
</div>
`;
